### DISCUSSION TITLE
Can't set topics via MQTT with MQTT Explorer

### DISCUSSION THREAD

**Discussion by skyynet** [john30/ebusd#683] [Q&A]
I installed the docker on my Synology and connected to our Vaillant heat pump via esera eBus copuple ethernet.
The topics are sent to Mosquitto and I can read them.
To start, I wanted to change the heating off/on (e.g. eco) with MQTT Explorer.

ebusd/mc/Mode holds "24;off;0;0;disabled;mixer;day" when heating is turned off and "24;eco;0;0;disabled;mixer;day" when turned on. I thought writing the opposite would turn heating on/off and used MQTT Explorer to publish the above string as RAW to the topic ebusd/mc/Mode. I also added "/set" to the back. Unfortunately nothing happens. I can see the value in MQTT Explorer changing, but not on the heat pump. What am I doing wrong? I'm quite sure, this is not a bug in ebusd but me not figuring this out. I can successfully switch a Wi-Fi relay board with MQTT Explorer, so in general this should work.

**Comment by gnagy** [comment]
I have a similiar question using an Ariston Genus One boiler, MQTT read works in Home Assistant and I can write with ebusctl -w but nothing happens from MQTT explorer.

I'm testing with these messages (different circuit for testing) based on traffic i've grabbed using a compatible Ariston room thermostat:
```
r,ariston_broadcast,z1_night_temp,,,fe,200e,6371,,s,UIN,10,°C,
w,ariston_write,z1_night_temp,,,fe,2020,6371,,s,UIN,10,°C
```
Writing with ebusctl works, value in boiler broadcast is updated.
```
ebusctl -s neutron w -c ariston_write z1_night_temp 16
logs:
ebusd_1  | 2022-10-30 11:23:02.204 [bus info] send message: 31fe2020046371a000
ebusd_1  | 2022-10-30 11:23:02.375 [main notice] write ariston_write z1_night_temp: done
```
The write topic shows up in MQTT explorer and I see updates from ebusd on it. However, when I try to publish updates from MQTT explorer nothing seems to happen in ebusd:
```
topic: ebusd/ariston_write/z1_night_temp
payload: {"0": {"name": "", "value": 16.0}}
```

I'm using stock mqtt-hassio.cfg with minor modifications. I tried to append `|^w` to `filter-direction` and remove `filter-direction` alltogether, but if I undertsand this filter correctly then the topic for `ariston_write.z1_night_temp` would not appear at all on mqtt.

Any thoughts on how to get MQTT control working?

**Reply by gnagy** [reply]
I got it working. The topic name for setting is `ebusd/ariston_write/z1_night_temp/set` and the payload is not JSON, but plain value `17.0`

**Reply by john30** [reply]
yes, currently the /set topics only accept the same input as "ebusctl -w", this is tracked by #588
